packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart

```diff
@@ -0,0 +1,320 @@
+  /// Creates a new [AndroidWebViewCookieManager].
+  AndroidWebViewController(PlatformWebViewControllerCreationParams params)
+      : super.implementation(params is AndroidWebViewControllerCreationParams
+            ? params
+            : AndroidWebViewControllerCreationParams
+                .fromPlatformWebViewControllerCreationParams(params));
+
+  AndroidWebViewControllerCreationParams get _androidWebViewParams =>
+      params as AndroidWebViewControllerCreationParams;
+
+  /// The native [android_webview.WebView] being controlled.
+  late final android_webview.WebView _webView = withWeakRefenceTo(this, (
+    WeakReference<AndroidWebViewController> weakReference,
+  ) {
+    return _androidWebViewParams.androidWebViewProxy.createAndroidWebView(
+      useHybridComposition: true,
+    );
+  });
+
+  /// The native [android_webview.FlutterAssetManager] allows managing assets.
+  late final android_webview.FlutterAssetManager _flutterAssetManager =
+      withWeakRefenceTo(this, (
+    WeakReference<AndroidWebViewController> weakReference,
+  ) {
+    return _androidWebViewParams.androidWebViewProxy
+        .createFlutterAssetManager();
+  });
+
+  final Map<String, AndroidJavaScriptChannelParams> _javaScriptChannelParams =
+      <String, AndroidJavaScriptChannelParams>{};
+
+  @override
+  Future<void> loadFile(
+    String absoluteFilePath,
+  ) {
+    final String url = absoluteFilePath.startsWith('file://')
+        ? absoluteFilePath
+        : 'file://$absoluteFilePath';
+
+    _webView.settings.setAllowFileAccess(true);
+    return _webView.loadUrl(url, <String, String>{});
+  }
+
+  @override
+  Future<void> loadFlutterAsset(
+    String key,
+  ) async {
+    final String assetFilePath =
+        await _flutterAssetManager.getAssetFilePathByName(key);
+    final List<String> pathElements = assetFilePath.split('/');
+    final String fileName = pathElements.removeLast();
+    final List<String?> paths =
+        await _flutterAssetManager.list(pathElements.join('/'));
+
+    if (!paths.contains(fileName)) {
+      throw ArgumentError(
+        'Asset for key "$key" not found.',
+        'key',
+      );
+    }
+
+    return _webView.loadUrl(
+      'file:///android_asset/$assetFilePath',
+      <String, String>{},
+    );
+  }
+
+  @override
+  Future<void> loadHtmlString(
+    String html, {
+    String? baseUrl,
+  }) {
+    return _webView.loadDataWithBaseUrl(
+      baseUrl: baseUrl,
+      data: html,
+      mimeType: 'text/html',
+    );
+  }
+
+  @override
+  Future<void> loadRequest(
+    LoadRequestParams params,
+  ) {
+    if (!params.uri.hasScheme) {
+      throw ArgumentError('WebViewRequest#uri is required to have a scheme.');
+    }
+    switch (params.method) {
+      case LoadRequestMethod.get:
+        return _webView.loadUrl(params.uri.toString(), params.headers);
+      case LoadRequestMethod.post:
+        return _webView.postUrl(
+            params.uri.toString(), params.body ?? Uint8List(0));
```

Comment:
You'll need to add:

```
// TODO(a14n): remove this import once Flutter 3.1 or later reaches stable (including flutter/flutter#104231)
// ignore: unnecessary_import
import 'dart:typed_data';
```

to this file to fix the legacy analyze CI failure (as we currently have in a bunch of other plugin code).

Reply:
Added the additional import.